misc/docs-linter/links.go

```diff
@@ -3,105 +3,73 @@ package main
 import (
 	"bufio"
 	"bytes"
-	"context"
 	"fmt"
-	"golang.org/x/sync/errgroup"
-	"io"
-	"mvdan.cc/xurls/v2"
-	"net/http"
+	"os"
+	"path/filepath"
+	"regexp"
 	"strings"
-	"sync"
 )
 
-// extractUrls extracts URLs from a file and maps them to the file
-func extractUrls(fileContent []byte) []string {
+// extractLocalLinks extracts links to local files given file content
+func extractLocalLinks(fileContent []byte) []string {
 	scanner := bufio.NewScanner(bytes.NewReader(fileContent))
-	urls := make([]string, 0)
+	links := make([]string, 0)
+	// Regular expression to match markdown links
+	re := regexp.MustCompile(`]\((\.\.?/.+?)\)`)
 
 	// Scan file line by line
 	for scanner.Scan() {
 		line := scanner.Text()
 
-		// Extract links
-		rxStrict := xurls.Strict()
-		url := rxStrict.FindString(line)
+		// Find all matches
+		matches := re.FindAllString(line, -1)
 
-		// Check for empty links and skip them
-		if url == " " || len(url) == 0 {
-			continue
-		}
+		// Extract and print the local file links
+		for _, match := range matches {
+			// Remove ]( from the beginning and ) from end of link
+			match = match[2 : len(match)-1]
+
+			// Remove markdown headers in links
+			pos := strings.Index(match, "#")
+
+			if pos != -1 {
+				match = match[:pos]
+			}
+
+			pos = strings.Index(match, " ")
 
-		// Look for http & https only
-		if strings.HasPrefix(url, "http://") || strings.HasPrefix(url, "https://") {
-			// Ignore localhost
-			if !strings.Contains(url, "localhost") && !strings.Contains(url, "127.0.0.1") {
-				urls = append(urls, url)
+			if pos != -1 {
+				match = match[:pos]
 			}
+
+			links = append(links, match)
 		}
 	}
 
-	return urls
+	return links
 }
 
-func lintLinks(fileUrlMap map[string][]string, ctx context.Context) error {
-	// Filter links by prefix & ignore localhost
-	// Setup parallel checking for links
-	g, _ := errgroup.WithContext(ctx)
-
-	var (
-		lock         sync.Mutex
-		notFoundUrls []string
-	)
-
-	for filePath, urls := range fileUrlMap {
+func lintLocalLinks(fileLocalLinkMap map[string][]string, docsPath string) error {
+	found := false
```

Comment:
```suggestion
	var found bool
```

The default value is false.